### ISSUE TITLE
extract defaults from a simulator object

### ISSUE THREAD

**Issue by bbolker** [canmod/macpan2#166]
`mp_default` takes a spec or simulator and returns the default values.

## User Issue

If we have the following objects (`si` is a model spec, `si_sim` is a simulator)

```r
library(macpan2)
si <- mp_tmb_model_spec(
    before = list(
        I ~ 1
      , S ~ N - I
    )
  , during = list(
        infection_rate ~ beta * S * I / N
      , S ~ S - infection_rate
      , I ~ I + infection_rate
    )
  , default = list(N = 100, beta = 0.25)
)
si_sim <- mp_simulator(si,
        time_steps = 50
      , outputs = c("I", "infection_rate")
    )
```

It's easy to extract/see the default parameter values for `si` (via `print(si)` or `si$default`)  but really hard to figure it out for the simulator ... presumably these objects still exist somewhere inside some environment, but Greg and I had a hard time finding them!

We can find the *values* in `si2$tmb_model$ad_fun()$env$data`, but (a) it's ridiculous to dig that deep and (b) the names have been lost at this point ...

This works but seems horrible:

```r
get_vals <- function(obj) {
    mm <- obj$matrix_names()
    res <- lapply(mm, obj$get$initial)
    names(res) <- mm
   return(res)
}
get_vals(si2)
```

Suggestions?

**Comment by stevencarlislewalker** [comment]
My sense is that this issue is a sign that calibration is not yet a real user-facing feature.  But what you wrote will work, as you have discovered. I can expand later.

**Comment by bbolker** [comment]
But this doesn't even seem to be calibration yet ... we just want to know what the default parameters for the *simulator* are ... ? (Presumably *simulation* is user-facing?)

**Comment by stevencarlislewalker** [comment]
Oh ... well the defaults should be the same as for the spec.  If they are not the same then there is a bug somewhere.

**Comment by bbolker** [comment]
I'm sure you're right, but the design logic is still eluding me (documentation might fix this).  I make a spec object, add defaults. Then I make it into a simulator object, but I can no longer easily retrieve the default parameters - do I have to go back to the corresponding spec object to get them? To my mind there should at least be an accessor available for everything I might reasonably want to find out about an object ...

**Comment by stevencarlislewalker** [comment]
closed with [66c7488](https://github.com/canmod/macpan2/commit/66c7488d6f77ff323ee70e1dce1f135fbee41e25)
